libc-bottom-half/cloudlibc/src/common/net.h

```diff
@@ -64,12 +66,18 @@ static inline int wasi_to_sockaddr(const struct __wasi_addr_port_t *restrict pee
     } else if (peer_addr->tag == __WASI_ADDRESS_FAMILY_INET6) {
       struct sockaddr_in6 addr6;
       addr6.sin6_family = AF_INET6;
-      addr6.sin6_flowinfo = 0;
-      addr6.sin6_scope_id = 0;
+      addr6.sin6_flowinfo = peer_addr->u.inet6.addr.flow_info1 << 16 & peer_addr->u.inet6.addr.flow_info0;
+      addr6.sin6_scope_id = peer_addr->u.inet6.addr.scope_id1 << 16 & peer_addr->u.inet6.addr.scope_id0;;
```

Comment:
Same

Reply:
Because putting a 32 bit value in the struct changes the alignment of the entire `__wasi_addr_port_t` struct to 4, and that breaks every compiled wasix app out there.

Reply:
Mmm, ok. You could have used the `packed` attribute also.